bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java

```diff
@@ -377,6 +384,31 @@ private BigtableOptions resolveLegacyOptions(BigtableOptions options) throws IOE
     return options;
   }
 
+
+  /**
+   * Return options with legacy input options, if any, resolved into currently supported options.
+   */
+  public synchronized BigtableClusterName getClusterName() throws IOException {
+    if (options.getClusterName() == null) {
+      try (BigtableClusterUtilities util =
+          BigtableClusterUtilities.forInstance(options.getProjectId(), options.getInstanceId())) {
+        ListClustersResponse clusters = util.getClusters();
+        Preconditions.checkState(clusters.getClustersCount() == 1,
+          String.format(
+            "Project '%s' / Instance '%s' has %d clusters. There must be exactly 1 for this operation to work.",
+            options.getProjectId(), options.getInstanceId(), clusters.getClustersCount()));
+        Cluster cluster = clusters.getClusters(0);
+        options = options.toBuilder()
```

Comment:
The reason you want to mutate "options" here so that the next time when this is called, we don't have to lookup for cluster again. I don't know if it is worth it. I would rather just 1) do this in constructor or 2) not mutating "options" and return cluster ID here.

Up to you.